Major Update - Python 3.6 Compatible + Various API Changes

DEKHTIARJonathan opened the pull request:
I merged in this PR the work of @cprieto PR #6 and @viktorbengtsson PR #5.
The plugin API has been changed in order to allow more flexibility.

Thanks to the work of @viktorbengtsson, it solves issue #4 

Part of the work has been inspired by work done by @deepakrb on https://github.com/deepakrb/Pelican-Read-Time.

This PR aims to merge the plugins taking the best out of the two worlds.

Documentation has been updated to allow you to test and merge.

viktorbengtsson committed:
Updated plugin to resolve bug with parsing internal links using {filename}
Changed signal from content_object_init to all_generators_finalized (see https://github.com/getpelican/pelican/pull/1616)
Changed read_time to operate on content instances in generators
Disabled READTIME_CONTENT_SUPPORT setting.

cprieto committed:
Fix warning in Pelican 3.6 and above related to content

Commit:
Merge branch 'pr/1'

Commit:
API Remastered - Architecture Changed

DEKHTIARJonathan committed:
Update and rename README.rst to README.md

DEKHTIARJonathan committed:
Update README.md

DEKHTIARJonathan committed:
Update README.md

deepakrb commented on README.md:
````diff
@@ -0,0 +1,76 @@
+
+***pelicanconf.py***
+```python
+READTIME_LANGUAGE_SUPPORT = 180
````
`READTIME_LANGUAGE_SUPPORT` seems a bit ambiguous, maybe something like `READTIME_WPM`?

DEKHTIARJonathan replied:
I have changed according to your suggestion, I will commit ASAP ;)

deepakrb commented on readtime.py:
```diff
@@ -1,99 +1,159 @@
+                            raise Exception("Missing wpm value for the language: %s" % key)
 
+                        if not isinstance(settings_wpm[key]['wpm'], int):
+                            raise Exception("WPM is not an integer for the language: %s" % key)
```
There's a mix of two python string formatting `%s` vs `'{}'.format()`. Maybe one would be better?

deepakrb commented:
Cheers for taking a look at my repo @DEKHTIARJonathan it'd be great if it could end up somewhere since it didn't end up on the main pelican plugins repo 👍

Commit:
Code Refactored

Commit:
Readme converted from MD to RST

DEKHTIARJonathan commented:
@deepakrb your remarks have been added in the commit 63531cb.
You are very welcome, you made a very nice job. I just took a lot of inspiration from your work.

Commit:
Tests Updated to be working with the new API

Commit:
Setup.py file refactored

DEKHTIARJonathan commented:
I just modified and improved the *setup.py* file.
I also updated the tests.py to be working with the new API

@JenkinsDev maybe it would be good after merging this PR to submit the project to the official pelican-plugin repository
